docs/proposals/openshift_templates.md

```diff
@@ -0,0 +1,436 @@
+  },
+  "objects": [
+    {
+      "kind": "Service",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "$(DATABASE_SERVICE_NAME)"
+      },
+      "spec": {
+        "ports": [
+          {
+            "name": "mongo",
+            "protocol": "TCP",
+            "targetPort": 27017
+          }
+        ],
+        "selector": {
+          "name": "$(DATABASE_SERVICE_NAME)"
+        }
+      }
+    },
+
+    {
+      "kind": "ReplicationController",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "$(DATABASE_SERVICE_NAME)",
+      },
+      "spec": {
+        "replicas": "$(REPLICA_COUNT)",
+        "selector": {
+          "name": "$(DATABASE_SERVICE_NAME)"
+        },
+        "template": {
+          "metadata": {
+              "creationTimestamp": null,
+              "labels": {
+                  "name": "$(DATABASE_SERVICE_NAME)"
+              }
+          },
+          "spec": {
+            "containers": [
+              {
+                "name": "mongodb",
+                "image": "docker.io/centos/mongodb-26-centos7",
+                "ports": [
+                  {
+                    "containerPort": 27017,
+                    "protocol": "TCP"
+                  }
+                ],
+                "env": [
+                  {
+                    "name": "MONGODB_USER",
+                    "value": "${MONGODB_USER}"
+                  },
+                  {
+                    "name": "MONGODB_PASSWORD",
+                    "value": "${MONGODB_PASSWORD}"
+                  },
+                  {
+                    "name": "MONGODB_DATABASE",
+                    "value": "${MONGODB_DATABASE}"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      }
+    }
+  ],
+  "parameters": [
+    {
+      "name": "DATABASE_SERVICE_NAME",
+      "description": "Database service name",
+      "value": "mongodb",
+      "required": true,
+      "type": "string"
+    },
+    {
+      "name": "MONGODB_USER",
+      "description": "Username for MongoDB user that will be used for accessing the database",
+      "generate": "expression",
+      "from": "user[A-Z0-9]{3}",
+      "required": true,
+      "type": "string"
+    },
+    {
+      "name": "MONGODB_PASSWORD",
+      "description": "Password for the MongoDB user",
+      "generate": "expression",
+      "from": "[a-zA-Z0-9]{16}",
+      "required": true,
+      "type" :"string"
+    },
+    {
+      "name": "MONGODB_DATABASE",
+      "description": "Database name",
+      "value": "sampledb",
+      "required": true,
+      "type" :"string"      
+    },
+    {
+      "name": "REPLICA_COUNT",
+      "description": "Number of mongo replicas to run",
+      "value": "1",
+      "required": true,
+      "type" :"int"      
+    }
+  ]
+}
+```
+
+### API Endpoints
+
+* **/processedTemplates** - when a template is POSTed to this endpoint, all parameters in the template are processed, values 
```

Comment:
Why not a subresource that would allow me to post a map of parameter values to a template and get back a List of the expanded resources?